### PR TITLE
Multiple Fixes

### LINKED ISSUE #9: LeaseDuration is currently mandatory
The xDhcpServerScope actually requires the LeaseDuration property to be defined or the resource execution fails, but the configuration doesn't  enforce this.
- The LeaseDuration property should be marked as 'Required' in the schema.mof
- Default the LeaseDuration to "00:08:00" (my preferred option).

### LINKED ISSUE #14: Test-DscConfiguration always returns false on DnsServerIPAddress and DnsDomain
I'm trying to set the DNS-Domain and DNS-Server Adresse with the xDhcpServerOption resource, but the test always returns false

I'm working with Windows 10 and PS v5 ProductionPreview on Server 2012r2

```powershell
AUSFÜHRLICH: [WIN-DC1]:                            [[xDhcpServerOption]192.168.18.0] Checking Dns server ip option ...
AUSFÜHRLICH: [WIN-DC1]:                            [[xDhcpServerOption]192.168.18.0] DNS server ip is not correct. Expected System.String[], actual
AUSFÜHRLICH: [WIN-DC1]:                            [[xDhcpServerOption]192.168.18.0] Setting DNS server ip option ...
AUSFÜHRLICH: [WIN-DC1]:                            [[xDhcpServerOption]192.168.18.0] DNS server ip option is set to System.String[].
AUSFÜHRLICH: [WIN-DC1]:                            [[xDhcpServerOption]192.168.18.0] Checking Dns domain name option ...
AUSFÜHRLICH: [WIN-DC1]:                            [[xDhcpServerOption]192.168.18.0] DNS domain name is not correct. Expected lab.local, actual
AUSFÜHRLICH: [WIN-DC1]:                            [[xDhcpServerOption]192.168.18.0] Setting DNS domain name option ...
AUSFÜHRLICH: [WIN-DC1]:                            [[xDhcpServerOption]192.168.18.0] DNS domain name option is set to lab.local.


AUSFÜHRLICH: [WIN-DC1]:                            [[xDhcpServerOption]192.168.18.0] Checking Dns server ip option ...
AUSFÜHRLICH: [WIN-DC1]:                            [[xDhcpServerOption]192.168.18.0] DNS server ip is not correct. Expected System.String[], actual
AUSFÜHRLICH: [WIN-DC1]: LCM:  [ BeendenTesten   ]  [[xDhcpServerOption]192.168.18.0] False in 0.1560 Sekunden.
```

here is the conf:
```powershell
xDhcpServerOtion '192.168.18.0'
{
    ScopeID = '192.168.18.0'
    AddressFamily = 'IPv4'
    Router = '192.168.18.4'
    DnsDomain = 'lab.local'
    DnsServerIPAdress = '192.168.18.18','192.168.18.37'
    Ensure = 'Present'
}
```

### LINKED ISSUE #21: LeaseDuration Bug - Incorrect duration set
If I set Lease Duration to '02:00:00' then on the DHCP server that translates to

"Days: 0 Hours: 2 Minutes: 0"

If I set Lease Duration to '48:00:00' then on the DHCP server that translates to 

Days: 48 Hours: 0 Minutes: 0

### PR DESCRIPTION
**Pull request by iainbrighton**
* Fixes #9
 * Adds test coverage for xDhcpServerScope
* Fixes #14
 * Changes comparisons to use the OptionIds rather than names which are localized
* Fixes #21
 * Updates readme, giving LeaseDuration examples
* Implements unit test template
 * Add-WindowsFeature RSAT-DHCP role installation into appveyor.yml

---
This change is [Reviewable](https://reviewable.io/reviews/powershell/xdhcpserver/22)

### CODE REVIEW

File: DSCResources/MSFT_xDhcpServerOption/MSFT_xDhcpServerOption.psm1 [hunk 1]
```diff
@@ -259,7 +259,7 @@ function ValidateResourceProperties
                 $checkPropertyMessage = $($LocalizedData.CheckPropertyMessage) -f 'Dns server ip'
                 Write-Verbose -Message $checkPropertyMessage
                 
-                $dnsServerIP = ($dhcpOption | Where-Object Name -like 'DNS Servers').value
+                $dnsServerIP = ($DhcpOption | Where-Object OptionId -eq 6).Value
```
Comment (TravisEz13) [inline DSCResources/MSFT_xDhcpServerOption/MSFT_xDhcpServerOption.psm1]: we always get the `optionid`?

File: README.md [hunk 2]
```diff
@@ -23,6 +23,8 @@ Please check out common DSC Resources [contributing guidelines](https://github.c
 * **Name**: Name of this DHCP Scope
 * **SubnetMask**: Subnet mask for the scope specified in IP address format
 * **LeaseDuration**: Time interval for which an IP address should be leased
+ * This should be specified in the following format: `Days.Hours:Minutes:Seconds`
```
Comment (TravisEz13) [inline README.md]: Could we write a helper in the module to convert a timespan to this format?

Comment (TravisEz13) [inline README.md]: you could open an issue for this.

**Comment by TravisEz13** [comment]
Mainly looks good to me.  Had a couple of questions.

**Comment by TravisEz13** [comment]
@iainbrighton still looking for answers to questions in the review

File: DSCResources/MSFT_xDhcpServerOption/MSFT_xDhcpServerOption.psm1 (hunk 1 above)
Comment (iainbrighton) [inline DSCResources/MSFT_xDhcpServerOption/MSFT_xDhcpServerOption.psm1]: @TravisEz13 Yes - where the option is set. Unlike the Name parameter, this is localised!

File: README.md (hunk 2 above)
Comment (iainbrighton) [inline README.md]: @TravisEz13 Not sure why we need a helper method? In hindsight, it should have just been implemented as an integer, specifying the number of minutes. We could have then leveraged the ConvertTo-TimeSpan/ConvertFrom-TimeSpan helper methods in the xActiveDirectory module.

File: DSCResources/MSFT_xDhcpServerOption/MSFT_xDhcpServerOption.psm1 (hunk 1 above)
Comment (TravisEz13) [inline DSCResources/MSFT_xDhcpServerOption/MSFT_xDhcpServerOption.psm1]: Thanks

File: README.md (hunk 2 above)
Comment (TravisEz13) [inline README.md]: @iainbrighton how about we just update the sample to
```PowerShell
configuration Sample_xDhcpsServerScope_NewScope
{
    Import-DscResource -module xDHCpServer
    xDhcpServerScope Scope
    {
        Ensure = 'Present'
        IPEndRange = '192.168.1.254'
        IPStartRange = '192.168.1.1'
        Name = 'PowerShellScope'
        SubnetMask = '255.255.255.0'
        LeaseDuration = ((New-TimeSpan -Minutes 8 ).ToString())
        State = 'Active'
        AddressFamily = 'IPv4'
    }
}
```
from
```PowerShell
configuration Sample_xDhcpsServerScope_NewScope
{
    Import-DscResource -module xDHCpServer
    xDhcpServerScope Scope
    {
        Ensure = 'Present'
        IPEndRange = '192.168.1.254'
        IPStartRange = '192.168.1.1'
        Name = 'PowerShellScope'
        SubnetMask = '255.255.255.0'
        LeaseDuration = '00:08:00'
        State = 'Active'
        AddressFamily = 'IPv4'
    }
}
```

**Comment by TravisEz13** [comment]
@iainbrighton One unresolved issue in reviewable

---

Reviewed 10 of 10 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/powershell/xdhcpserver/22#-:-KED4X_I8f1KPWRReza0:28968224)*

File: README.md (hunk 2 above)
Comment (iainbrighton) [inline README.md]: That's a good idea - it's certainly more explicit what the intended timespan is. I'll update the example documentation.

Comment (iainbrighton) [inline README.md]: Done!

**Comment by TravisEz13** [comment]
Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xdhcpserver/22)*

**Pull request merged**